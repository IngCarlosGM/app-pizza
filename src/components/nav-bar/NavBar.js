import React from 'react';
import { useDispatch } from 'react-redux';
import { startLogout } from '../../actions/auth';

export const NavBar = ({ history }) => {

    const dispatch = useDispatch();

    const handleLogout = () => {
        dispatch( startLogout() );
        history.replace("/auth/login");
    }

    return (
        <nav className="align-center-right">
            <img 
                className="img_responsive size-icon-logout" 
                src={`../assets/ic_contrasena.png`} 
                alt="Ã­cono candado"
            />
            <span 
                className="text-logout pointer"
                onClick={ handleLogout }
            >Salir</span>
        </nav>
    )
}