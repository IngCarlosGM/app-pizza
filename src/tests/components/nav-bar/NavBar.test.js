import { mount } from "enzyme";
import { Provider } from "react-redux";
import { NavBar } from '../../../components/nav-bar/NavBar';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initialState = {
    auth: {
        name: 'Pepito Perez',
        logged: true
    },
    ui: {
        loading: false,
        msgError: null
    },
    dashboard: {
        
    }
}

let store = mockStore(initialState);

describe('Testing NavBar component', () => {

    const replace = jest.fn();

    const history = {
        replace
    }
   
    const wrapper = mount(
        <Provider store={ store } >
            <NavBar 
                history= { history }
            />
        </Provider>
    );

    test('should render correctly', () => {
        expect( wrapper ).toMatchSnapshot();
    });
    
    test('should to have a img', () => {
        expect( wrapper.exists('img') ).toBe( true );
    });

    test('should to have a span', () => {
        expect( wrapper.exists('span') ).toBe( true );
    });

    test('should to call the replace function', () => {
        wrapper.find('span').at(0).simulate('click');
        expect( replace ).toHaveBeenCalledTimes( 1 );
    });
});
