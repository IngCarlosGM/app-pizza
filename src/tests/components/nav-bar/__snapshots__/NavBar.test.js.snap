// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing NavBar component should render correctly 1`] = `
<nav
  className="align-center-right"
>
  <img
    alt="Ã­cono candado"
    className="img_responsive size-icon-logout"
    src="../assets/ic_contrasena.png"
  />
  <span
    className="text-logout pointer"
    onClick={[Function]}
  >
    Salir
  </span>
</nav>
`;
